### PR TITLE
Docs Update -- Developer Guide

### LINKED ISSUE #46: Library Contributor Guide
Create docs to help people developing contributions to the library.

* Technical stuff (how the library works, how to structure code)
* Logistical stuff (how contributions are validated)

### LINKED ISSUE #271: Docs Update -- Review Process
Rework Tuesday [meeting notes](https://docs.google.com/document/d/1X04N7SMSaVTt4BHsuKbPOOGBeG-FYKUY59HCxTzVgBM/edit) and add them to the "Review Process" section of the Developer Guide.

### LINKED ISSUE #291: Issue Templates
Issue templates can help route users to the right places, and we can link an issue template directly from the docs.

### PR DESCRIPTION
**Pull request by Shoeboxam**
Closes #46 
Closes #291
Closes #271 

This PR fleshes out the developer guide and contributor/reviewer process.

Also adds some issue templates that are linked together with the developer guide docs. Live demo of the templates:
https://github.com/Shoeboxam/test-issue-templates/issues/new/choose

I have found issue templates grating in the past-- there are oftentimes many irrelevant fields that cause issues to be strangely worded. So I tried to keep the templates simple.

The hope here was that these would
- give a soft reminder to use the security email if relevant
- help guide the user along the contribution process
- direct some more attention towards discussions, including feature requests
- enrich the links in the developer guide

### CODE REVIEW

**Comment by pdurbin** [comment]
Just a quick note to say I love the use of issue templates!

File: docs/source/developer/code-structure.rst [hunk 1]
```diff
@@ -1,14 +1,67 @@
...
+You may find the more general row transform proof useful.
+If your transformation is a row-transform, you can lean on this proof-
+you only need to prove that your transformation meets the supporting requirements.
+
+FFI Wrapper
+===========
+Any constructor that does not have an FFI wrapper will not be available in bindings languages.
+At the same time, we acknowledge that writing FFI wrappers can be tricky.
+We are working on more automated tooling to generate FFI wrappers, but in the meantime,
+it is likely a core library developer will kindly offer to write FFI wrappers for you,
```
Comment (andrewvyrros) [inline docs/source/developer/code-structure.rst:62]: Ummm... How about, "but in the meantime, a core developer can work with you to write the FFI wrappers, if you write the constructor."

File: docs/source/developer/code-structure.rst [hunk 2]
```diff
@@ -1,14 +1,67 @@
...
+#. The proof documents should be located in the same folder as your constructor function.
+   At the moment there are many proofs undergoing review `in an adjacent repository <https://github.com/opendp/whitepapers/pulls>`_.
+   We will be moving these proof files to be adjacent to the code in the near future,
+   but for now they may provide a useful template and definition base for your own proofs.
+#. The linked repository contains two definitions files you should reference.
+   Going forward we intend to use these shared definitions for all proofs, for consistency.
+
+You may find the more general row transform proof useful.
+If your transformation is a row-transform, you can lean on this proof-
+you only need to prove that your transformation meets the supporting requirements.
```
Comment (andrewvyrros) [inline docs/source/developer/code-structure.rst:55]: Em dash

```suggestion
If your transformation is a row-transform, you can lean on this proof—you
only need to prove that your transformation meets the supporting requirements.
```

File: docs/source/developer/contribution-process.rst [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+After making your pull request, please check back for feedback from the core developers.
+
+On the OpenDP side, the review process for the proof goes as follows:
+
+#. Reviewers are assigned.
+#. Reviewer confirms the privacy claims of the proof.
+#. Reviewer validates the pseudo-code against the proof.
+#. Reviewer validates the Rust code against the pseudo-code.
+
+Please be patient- this process may take several iterations as issues may be raised in the review process.
```
Comment (andrewvyrros) [inline docs/source/developer/contribution-process.rst:49]: Em dash

```suggestion
Please be patient—this process may take several iterations as issues may be raised in the review process.
```

File: docs/source/developer/logistics.rst [hunk 4]
```diff
@@ -89,3 +94,27 @@ Building the Docs
...
+The benefit to using Intellij IDEA is that the core developers use it,
+which makes it possible for one of us to actually join your IDE with the `CodeWithMe Plugin <https://www.jetbrains.com/code-with-me/>`_,
+and talk through issues.
+
+A few notes on Intellij IDEA:
+
+* Both the Intellij IDEA community edition and the CodeWithMe plugin are free
+* Be sure to open the project at the root of the git repository
+* Be sure to install the Python and Rust plugins for interactivity
+* Be sure to "attach" the Cargo.toml in the red banner the first time you open a rust source file
```
Comment (andrewvyrros) [inline docs/source/developer/logistics.rst:118]:
```suggestion
* Be sure to "attach" the Cargo.toml in the red banner the first time you open a Rust source file
```

File: docs/source/developer/intro.rst [hunk 5]
```diff
@@ -4,12 +4,91 @@ Introduction
...
     :local:
 
-Purpose
-=======
 
 This guide provides information for those interested in participating in OpenDP development.
 
 If your focus is using OpenDP to build your own applications, please consult the :doc:`../user/index`.
 
+How to Get Involved
```
Comment (andrewvyrros) [inline docs/source/developer/intro.rst:12]: I like the way you restructured this, with enumerating the ways to contribute, each with pointers how to get started.

File: docs/source/developer/code-structure.rst [hunk 6]
```diff
@@ -1,14 +1,67 @@
...
+The requirements to be met for the vetting process are numbered for each component.
+
+Constructor Function
+====================
+
+The user guide has a relevant section explaining :ref:`constructors <constructors>`.
+
+#. The constructor is the only mandatory component to implement!
+#. If the proof is not written, then your work must be marked by the ``contrib`` flag.
+   In rust, you just need to add an annotation ``#[cfg(feature="contrib")]`` above an element to mark it.
```
Comment (andrewvyrros) [inline docs/source/developer/code-structure.rst:19]:
```suggestion
   In Rust, you just need to add an annotation ``#[cfg(feature="contrib")]`` above an element to mark it.
```

File: docs/source/developer/code-structure.rst [hunk 7]
```diff
@@ -1,14 +1,67 @@
...
+
+#. Your proof should show the following:
+
+    #. That the function, when evaluated on any element in the input domain, emits a value in the output domain.
+    #. That the relation always returns false if the function is not (``d_in``, ``d_out``)-close for all ``d_in`` and ``d_out``.
+    #. That your choices of metrics/measures are compatible with your domains.
+
+#. Your proof should include pseudocode.
+   This pseudocode is necessary for the vetting process:
+   We will need to check that the rust and pseudocode are isomorphic.
```
Comment (andrewvyrros) [inline docs/source/developer/code-structure.rst:45]:
```suggestion
   We will need to check that the Rust and pseudocode are isomorphic.
```

File: docs/source/developer/intro.rst [hunk 8]
```diff
@@ -4,12 +4,91 @@ Introduction
...
+Please `open a new issue <https://github.com/opendp/opendp/issues/new?template=new-contribution.md>`__ to take ownership of a piece of documentation.
+
+You can follow the :ref:`contribution-process` from here on.
+
+Add Tests
+---------
+We need your help to make OpenDP more stable by adding tests.
+
+* python: `pytest unit and integration tests <https://github.com/opendp/opendp/tree/main/python/test>`_
+* rust: in-file testing modules (`core example <https://github.com/opendp/opendp/blob/ead11c5dbadfb17062182da6799f400888e66cef/rust/opendp/src/trans/count/mod.rs#L121-L182>`_, `ffi example <https://github.com/opendp/opendp/blob/ead11c5dbadfb17062182da6799f400888e66cef/rust/opendp-ffi/src/trans/resize.rs#L53-L93>`_)
```
Comment (andrewvyrros) [inline docs/source/developer/intro.rst:72]:
```suggestion
* Python: `pytest unit and integration tests <https://github.com/opendp/opendp/tree/main/python/test>`_
* Rust: in-file testing modules (`core example <https://github.com/opendp/opendp/blob/ead11c5dbadfb17062182da6799f400888e66cef/rust/opendp/src/trans/count/mod.rs#L121-L182>`_, `ffi example <https://github.com/opendp/opendp/blob/ead11c5dbadfb17062182da6799f400888e66cef/rust/opendp-ffi/src/trans/resize.rs#L53-L93>`_)
```

**Review by andrewvyrros — CHANGES_REQUESTED** [review]
More great stuff! I like how you've guided people in the ways to get involved, and the issue templates are nice.

Thanks for breaking up my long lines!

Just some minor tweaks suggested.

File: .github/ISSUE_TEMPLATE/new-contribution.md [hunk 9]
```diff
@@ -6,7 +6,7 @@ about: Solicit feedback for a contribution you'd like to make
 How exciting!
 
 Please review the following link for information to include in your issue:
-https://docs.opendp.org/en/develop/developer/intro.html#code-and-proof-contributions.html
+https://docs.opendp.org/en/develop/developer/intro.html#code-and-proof-contributions
```
Comment (pdurbin) [inline .github/ISSUE_TEMPLATE/new-contribution.md:9]: I'm confused. Do we really want the "develop" branch here? Shouldn't it be "stable" or "latest"?

**Review by pdurbin — COMMENTED** [review]
I think I found a bad link.

File: .github/ISSUE_TEMPLATE/new-contribution.md (hunk 9 above)
Comment (Shoeboxam) [reply .github/ISSUE_TEMPLATE/new-contribution.md]: Nope, we don't want "develop". Crafting these urls by hand is error-prone. Thanks for catching this!

**Review by andrewvyrros — APPROVED** [review]
Excellent! This should be a big help for guiding people from the Community Meeting.

**Pull request merged**